doc/loader.rst

```diff
@@ -5,128 +5,539 @@ MIG LOADER
 .. sectnum::
 .. contents:: Table of contents
 
-**Note**: mig-loader is a work in progress component and is not complete. This
-document is intended to summarize the concept and planned approach, and will be
-updated once additional progress is made.
-
 Overview
 --------
-In normal scenarios, the MIG agent is deployed using some form of package, for
-example RPM or deb. When the agent needs to be updated, the older agent package
-is removed from the system and a new package is installed.
+The MIG loader is the component in MIG that can look after keeping agents up
+to date on systems. More specifically, this involves periodically checking if
+new agent code is available, making sure it's signed and safe to install, and
+handling automatic agent/configuration upgrades.
+
+In environments with a mature configuration management setup (such as via
+puppet or ansible) MIG loader may not be required. The configuration management
+system can periodically push updates out to required systems.
+
+In other environments, the loader can help keep things up to date. As an example
+this could be in cases where you want to keep MIG up to date on workstations that
+are not centrally managed. Even in cases with a configuration management system in
```

Comment:
"keep things up to date, for example on workstations and laptops that are not centrally managed."